src/translator/byte_array_util.cpp

```diff
@@ -89,13 +89,17 @@ AlignedMemory loadFileToMemory(const std::string& path, size_t alignment){
 }
 
 AlignedMemory getModelMemoryFromConfig(marian::Ptr<marian::Options> options){
-    auto models = options->get<std::vector<std::string>>("models");
-    ABORT_IF(models.size() != 1, "Loading multiple binary models is not supported for now as it is not necessary.");
-    marian::filesystem::Path modelPath(models[0]);
-    ABORT_IF(modelPath.extension() != marian::filesystem::Path(".bin"), "The file of binary model should end with .bin");
-    AlignedMemory alignedMemory = loadFileToMemory(models[0], 256);
-    ABORT_IF(!validateBinaryModel(alignedMemory, alignedMemory.size()), "The binary file is invalid. Incomplete or corrupted download?");
-    return alignedMemory;
+  auto models = options->get<std::vector<std::string>>("models");
+  ABORT_IF(models.size() != 1, "Loading multiple binary models is not supported for now as it is not necessary.");
+  marian::filesystem::Path modelPath(models[0]);
+  ABORT_IF(modelPath.extension() != marian::filesystem::Path(".bin"), "The file of binary model should end with .bin");
+  AlignedMemory alignedMemory = loadFileToMemory(models[0], 256);
+  bool check = options->get<bool>("check-bytearray",true);
+  if (check) {
+    ABORT_IF(!validateBinaryModel(alignedMemory, alignedMemory.size()),
+             "The binary file is invalid. Incomplete or corrupted download?");
+  }
+  return alignedMemory;
```

Comment:
This path is not fired when mozilla calls through API, we want to give them the ability to toggle check-bytearray as well, yes? Can't we have the check/validate code in one place, i.e inside Service consistent with where shortlist is being checked as well?

This function is so far only used in service-cli right?

Reply:
OK. I will move to Service.